### ISSUE TITLE
Pop growth in set of countries in 7.78

### ISSUE THREAD

**Issue by wverhagen** [PardeeCenterDU/IFs-Issues-Tracking#118]
Population growth rates in a subset of countries initially are very high, and jump up from historic growth patterns. There is then a strong transient between the initial year and the longer-term forecast.

Here is an example of Bahrain, Israel and Uganda. All have high immigration rates in 2017.

Thanks to Audrey for finding this issue 

![image](https://user-images.githubusercontent.com/18350553/139475370-5b8c71cf-cec0-413d-9630-b3aae83160ec.png)

A second problematic point is that these changes do not show up in the population graphs themselves so we have a popr that is not related to the resulting population graph. see also other issue on disjoint between population and popr calculation

![image](https://user-images.githubusercontent.com/18350553/139475478-acdf07be-347a-44b9-8656-dc75f0632cde.png)

I gave it a prio 4 for now. If it is only these countries we might be fine and can lower priority to 3 (it still needs to be fixed though). 
If there are more general issues with pop dynamics we have a bigger issue for the release (dave, jose feel free to amend the priority)

Steps for now are:
Identify why popr is so high, and then drops.
Identify why popr and growth in population are not alligned

**Comment by PardeeCenterIFs** [comment]
The first 2, Israel and Bahrain have a high POPR because of MIGRATE.
Uganda in version 7.78 has a POPR = 4.22, so the first graph must have been prepared with a different version.

![image](https://user-images.githubusercontent.com/33665510/140183317-da8cd919-b454-4f08-8fcc-7c5fa3bb2c53.png)

**Comment by PardeeCenterIFs** [comment]
Regarding the second question, I do think POPR and Population Growth are in line, here is a graph of POP, when going to Percent in a Table and Line graph.

![image](https://user-images.githubusercontent.com/33665510/140186615-2fe328bc-3cc1-40d1-a028-a64dbf91b3cd.png)

**Comment by wverhagen** [comment]
Providing some additional information. Below are the population growth rates for Uganda calculated in two ways
1) the first table gives the POPR value
2) the second tables gives the percent change in population (pop series) from one year to the previous

In theory population growth calculated using method 2 should be the same as pop growth calculated using method 1. 
It clearly is not. Even when accounting for a 1 year shift. The popr is at 4.2(2017), whereas the population growth from the population series is at 4.1 (2018).

Likely there is an issue here with when and how each of these series are calculated. But we should:
1) be able to understand why they are different
2) ideally update the order in the code so that they are indeed 1-on-1 the same

Because right now I don't know what the yearly pop growth is in a country.  

![image](https://user-images.githubusercontent.com/18350553/140826192-5754e52b-9d37-4e1a-9661-c30a61efcb48.png)

![image](https://user-images.githubusercontent.com/18350553/140826292-4f2524cb-d1a6-4c0b-b143-aa0e7a511763.png)

**Comment by wverhagen** [comment]
On the first question. Agreed that population growth in Bahrain and Israel comes from high migration in rates.

But that is not enough I think. 1) these popr drives igdpr very strongly so it results in a long-term boost to economic growth
2) we need to find out whether this is a finding (migration was really high in that particular for that country) or a data anomaly

So I'm sorry if I had misstated my initial question. It's not only why is pop growth so high for this country, but also is it correct that pop growth from migration is so high and then drops off so suddenly (my current read is it's actually wrong).

For example graph from world bank has israel in 2017 at 1.9 percent population growth. the 7% pop growth in IFs is higher than any pop growth in Israel since the 1960s. 
Bahrain shows more variation but only at 4.6 percent growth

![image](https://user-images.githubusercontent.com/18350553/140827502-71668be4-22d1-4921-8de8-203c46b59c91.png)
![image](https://user-images.githubusercontent.com/18350553/140827588-0a172ea8-77d7-4a1a-bf38-24c8d1a64f92.png)

If it's not correct we should update the data/model for initial conditions.
If it correct, we should update igdpr calc so that it is not affected by a single year outlier that happens to occur in 2017.

**Comment by PardeeCenterIFs** [comment]
Accounting for a 1 year shift. there's only difference in year 1, all other years are fine.
Why it doesn't match in year 1? because we compute POPR, using CDR with preliminary values.
POPR in year 1 is computed in ifsPOP(1), which is the first procedure of the run.
POPR is then used ifsECO(1) to compute IGDPRCor. IFsECO(1) is the 2nd procedure of the run.
CDR is recomputed in ifsPOP(2), so the real CDR doesn't make it into POPR calculation, so POPR in year 1 is an aproximation.

This re-calculation of CDR in ifsPOP(2) adds the effects of Starvation, War Deaths, Plague (AIDS), and substracts the efffects of Technological Advance, Heathcare Spending.

The code has a comment after the first calculation of CDR where it mentions this issue:

    'This can be quite different from the CDR computed later in the first year
    '   Thus anything that uses it or POPR before then is subject to a first
    '   year transient; this calculation of dth should be refined to be more
    '   like Deaths later on

**Comment by PardeeCenterIFs** [comment]
" is it correct that pop growth from migration is so high and then drops off so suddenly (my current read is it's actually wrong)", I think that this is a migration question and should be a different issue. I think that when we made the move from Net Migration initialization to Gross Migration we might have been too optimistic on the quality of our Gross data, and estimations.

**Comment by wverhagen** [comment]
I will make a second issue on migration

**Comment by wverhagen** [comment]
On the first issue. I don't fully agree it is a initialization issue. I think it is two issues:
1) it's an initialization issue
2) even if we account for the 1-year shift it still does not matter

a couple of important things:
- this is uganda I have not checked other countries but the differences between years might be bigger
- it starts off as a 4th decimal but is actually a 3rd decimal difference by 2050. I think that is significant is the pop growth rate 1.78 or 1.77 in year x? 

My proposal would be the following:
We need to be able to answer two basic question:
1) what is the population growth in year 1 in country x?
We need to write down somewhere what the actual correct pop growth rate is for the base year. is it popr or should we ignore that. This is crucial for any report we publish. I think we should discuss this in  the modelling pod in two weeks? If we don't have a answer then, we need to understand what's right and wrong. Otherwise I and others will publish report with the wrong population growth.

the second question then is "what is the population growth in country x in year x?
@dkbohl: could we have a resource or data team RA do the following. Pull all population and popr data for all countries. Calculate the population growth for both series for each country over time, and compare them. That way we have a sense of does it just not match for uganda or for more countries, and how big is the difference. My general rule of thumb would be that everything 3rd decimal or bigger is significant because I cannot answer anymore in a report (often 1 or two decimals precision) what the pop growth is?
We could then also bring that to the modelling pod two weeks from now

**Comment by PardeeCenterIFs** [comment]
You’re right Willem, there’s an issue, I found (thanks to your US example) that we were not passing migrants in the age category of 1-4, so if that category was 5% of total migration, we were loosing 5% of those migrants, so in the case of the US that’s why POP was slightly lower than POPR. I have added the code to include those extra migrants in the 1-4 age category, and the fix will be included in the next consolidation.

On the POPR in year 1, I agree, we should update it, I'll include it at the top of the agenda for our next modeling Pod.
